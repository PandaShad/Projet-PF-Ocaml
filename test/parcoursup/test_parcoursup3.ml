open Test_parcoursup_utils
let session = nouvelle_session ()

let () = 
  ajoute_candidat session ~nom_candidat:"Adam";
  ajoute_candidat session ~nom_candidat:"Béatrice";
  ajoute_formation session ~nom_formation:"Université Côte d'Azur" ~capacite:1;
  ajoute_formation session ~nom_formation:"Université de Toulon" ~capacite:2;
  ajoute_voeu session 
    ~rang_repondeur:None
    ~nom_candidat:"Adam" 
    ~nom_formation:"Université Côte d'Azur";
    ajoute_voeu session 
    ~rang_repondeur:None
    ~nom_candidat:"Adam" 
    ~nom_formation:"Université de Toulon";
    ajoute_voeu session 
    ~rang_repondeur:(Some 1) 
    ~nom_candidat:"Béatrice" 
    ~nom_formation:"Université Côte d'Azur";
    ajoute_voeu session 
    ~rang_repondeur:(Some 2) 
    ~nom_candidat:"Béatrice" 
    ~nom_formation:"Université de Toulon";
  ajoute_commission session 
    ~nom_formation:"Université Côte d'Azur" 
    ["Adam";"Béatrice"]; 
    ajoute_commission session 
    ~nom_formation:"Université de Toulon" 
    ["Adam";"Béatrice"]; 
  reunit_commissions session;
  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Béatrice";
  affiche_propositions_en_attente session "Béatrice";
  renonce session 
    ~nom_candidat:"Adam" 
    ~nom_formation:"Université Côte d'Azur";
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Béatrice";
  affiche_propositions_en_attente session "Béatrice";
  nouveau_jour session;
  affiche_voeux_en_attente session "Adam";
  affiche_propositions_en_attente session "Adam";
  affiche_voeux_en_attente session "Béatrice";
  affiche_propositions_en_attente session "Béatrice"
