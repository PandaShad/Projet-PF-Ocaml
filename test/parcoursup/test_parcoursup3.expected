nouvelle session
ajoute candidat Adam
ajoute candidat Béatrice
ajoute formation Université Côte d'Azur (1 places)
ajoute formation Université de Toulon (2 places)
ajoute voeu de Adam pour Université Côte d'Azur
ajoute voeu de Adam pour Université de Toulon
ajoute voeu de Béatrice pour Université Côte d'Azur (rang=1)
ajoute voeu de Béatrice pour Université de Toulon (rang=2)
commission de classement de Université Côte d'Azur : [Adam; Béatrice]
commission de classement de Université de Toulon : [Adam; Béatrice]
réunit commissions
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université Côte d'Azur, Université de Toulon}
Voeux en attente de Béatrice : {Université Côte d'Azur(0)}
Propositions en attente de Béatrice : {Université de Toulon}
Adam renonce à Université Côte d'Azur
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université de Toulon}
Voeux en attente de Béatrice : {Université Côte d'Azur(0)}
Propositions en attente de Béatrice : {Université de Toulon}
nouveau jour
Voeux en attente de Adam : {}
Propositions en attente de Adam : {Université de Toulon}
Voeux en attente de Béatrice : {}
Propositions en attente de Béatrice : {Université Côte d'Azur}
